auth_provider/remote_action_generators.py

```diff
@@ -0,0 +1,15 @@
+from ast import Str
+from typing import Callable, Optional
+
+from .models import User
+
+# Holds the name, and link of an action
+RemoteAction = Optional[tuple[str, str]]
+# Generates a UserAction when called with current_user and target_user
+RemoteActionGenerator = Callable[[User, Str], RemoteAction]
```

Comment:
Is there a reason we use `Str` rather than `str` here?

Reply:
```suggestion
RemoteActionGenerator = Callable[[User, str], RemoteAction]
```

Reply:
Good catch! I was so overwhelmed and made too many typos lol🥲

Reply:
hahaha I feel the vibe